mixer/adapter/circonus/circonus.go

```diff
@@ -0,0 +1,171 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package circonus
+
+import (
+	"context"
+	"fmt"
+	cgm "github.com/circonus-labs/circonus-gometrics"
+	"github.com/hashicorp/go-multierror"
+	"time"
+
+	"istio.io/istio/mixer/adapter/circonus/config"
+	"istio.io/istio/mixer/pkg/adapter"
+	"istio.io/istio/mixer/template/metric"
+)
+
+type (
+	builder struct {
+		adpCfg      *config.Params
+		metricTypes map[string]*metric.Type
+	}
+
+	handler struct {
+		cm          cgm.CirconusMetrics
+		metricTypes map[string]*metric.Type
+		env         adapter.Env
+		metrics     map[string]config.Params_MetricInfo_Type
+	}
+)
+
+// ensure types implement the requisite interfaces
+var _ metric.HandlerBuilder = &builder{}
+var _ metric.Handler = &handler{}
+
+///////////////// Configuration-time Methods ///////////////
+
+// adapter.HandlerBuilder#Build
+func (b *builder) Build(ctx context.Context, env adapter.Env) (adapter.Handler, error) {
+
+	cmc := &cgm.Config{}
+	cmc.CheckManager.Check.SubmissionURL = b.adpCfg.SubmissionUrl
```

Comment:
Initialize with struct init syntax instead?

Reply:
Done, and additional fmt/linter/doc changes have been pushed.